prototype2/common/EFUArgs.cpp

```diff
@@ -140,7 +165,8 @@ EFUArgs::Status EFUArgs::parseFirstPass(const int argc, char *argv[]) {
   } catch (const CLI::ParseError &e) {
     CLIParser.exit(e);
   }
-  if ((*HelpOption and not *DetectorOption) or (not *HelpOption and not *DetectorOption)) {
+  if ((*HelpOption and not*DetectorOption) or
+      (not*HelpOption and not*DetectorOption)) {
```

Comment:
if (not *DetectorOption)

Reply:
Nope, the code is correct as is. Lets print out the truth table.

```
Help flag | Det. flag
---------------------
True        True      -> Defer printing help until EFUArgs::parseSecondPass()
True        False     -> Print help and exit
False       True      -> Defer (possibly) printing help until EFUArgs::parseSecondPass()
False       False     -> Print help and exit
```

Reply:
![screen shot 2018-06-08 at 13 00 16](https://user-images.githubusercontent.com/4747406/41154900-040b54a8-6b1c-11e8-987e-92f32d6dc9d8.png)

I disagree - and so does randomly chosen online logic simplifier.

Reply:
Looking at it one last time, the code __is correct__. However, despite your first comment I had not noticed that the Help-flag did nothing.
With that said, seeing as the code __is correct__ and that the current version better explains what is happening than `not *DetectorOption` and that we have no performance to worry about, I suggest that we leave it as is.

Reply:
hmm... sure we can leave it as is. 

But to me it actually sends the wrong message: I first thought is had something to do with help option which it does not. It would still be 'correct' if we added
and (true != false) to the if statement but you would probably not argue for doing that. This is equivalent to dead code. 

Feel free to merge with or without changing this, but be prepared to hear about this again :-)